src/components/common/ColonyHome/ColonyFundingWidget/TokenBalanceItem.tsx

```diff
@@ -2,35 +2,28 @@ import React from 'react';
 
 import Numeral from '~shared/Numeral';
 import IconTooltip from '~shared/IconTooltip';
-// import { TokenBalancesForDomainsQuery } from '~data/index';
 import { getTokenDecimalsWithFallback } from '~utils/tokens';
 
 import { Token } from '~types';
 
 import styles from './ColonyFundingWidget.css';
 
 interface Props {
-  // currentDomainId: number;
   token: Token;
   isTokenNative?: boolean;
   isNativeTokenLocked?: boolean;
+  balance: string;
 }
 
 const displayName = 'common.ColonyHome.ColonyFundingWidget.TokenBalanceItem';
 
 const TokenBalanceItem = ({
-  // currentDomainId,
   token: { decimals, symbol },
   isTokenNative,
   isNativeTokenLocked,
-}: Props) => {
-  // const domainBalance = balances.find(
-  //   ({ domainId }) => domainId === currentDomainId,
-  // );
-  // const balance = domainBalance && domainBalance.amount;
-  const balance = 0;
-
-  return typeof balance === 'undefined' ? null : (
+  balance,
+}: Props) =>
+  typeof balance === 'undefined' ? null : (
```

Comment:
Balance is defined as `string` but here we're checking if it's undefined... Something wrong with types along the way?

Reply:
Digging deeper, it can indeed be undefined (since it's an array element at an arbitrary index), in which case I think we should change the prop type